source/contributed/caching-overview.md

```diff
@@ -0,0 +1,59 @@
+title: Caching Overview
+---
+
+Screeps is a performance centric game, as the better your performance the more you can accomplish in each tick. Caching is an extremely important element in performance optimization, and Screeps provides it's own unique opertunities and challenges when it comes to cache.
+
+Caching is, at it's core, a simple concept- by saving the results of expensive code future calls are less expensive. Caching also has a secondary benefit that can be exploited, in that it allows functions to returns results even if they shouldn't be able to. As an example caching the results of a pathfinding operation will let creeps reuse that optimal path even if visibility is lost to room along that path, which in turn provides both improved performance and resiliency.
+
+## Datastores
+
+### Memory
+
+The most common place to store cached data is in `Game.memory` (or just `Memory`). This is the only option for true persistance- anything stored in `Memory` is going to stay there until removed, so if a value is extremely expensive or most be saved this is the place to put it.
+
+There are two major drawbacks `Memory`-
+
+*   Space in `Memory` is limited to 2048kb of space.
+*   JSON.parse is run on the `Memory` string each tick it is accessed, which is more expensive the more data that is stored.
+
+For these reasons it makes sense to limit what is placed in `Memory`.
+
+
+### Global
+
+The `global` object persists between ticks as a byproduct of the [game loop]('/game-loop.html'). What this means is that players have access to an ephemeral storage- the `global` object- that will last between ticks.
+
+In this example a top level variable is defined in a module (`expensive.js`) and gets added to the global cache when the file is loaded with `require`. After the first time it the function is called the variable will be populated for future calls:
+
+    let runExpensiveCodeResults = false
+    function runExpensiveCode() {
+        if(!runExpensiveCode) {
+            runExpensiveCode = someExpensiveCode()
+        }
+        return runExpensiveCode
```

Comment:
Thanks for the catch- I went ahead and changed the variable name.

With regards to backticks- the rest of the site documentation is using indentations, which is why I went that route. If we want to stay consistent we should use indents or change the rest of the code on the site.